Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/details/VaccinationDetailsViewModel.kt

```diff
@@ -1,62 +1,74 @@
-    private val vaccinationRepository: VaccinationRepository,
     @Assisted private val vaccinationCertificateId: String,
+    private val qrCodeGenerator: QrCodeGenerator,
+    vaccinationRepository: VaccinationRepository,
     dispatcherProvider: DispatcherProvider,
-) : CWAViewModel() {
+) : CWAViewModel(dispatcherProvider) {
 
-    val vaccinationCertificate = vaccinationRepository.vaccinationInfos.map {
-        findVaccinationDetails(it)
-    }.asLiveData(context = dispatcherProvider.Default)
+    private var qrCodeText: String? = null
+    private val mutableStateFlow = MutableStateFlow<Bitmap?>(null)
+    val qrCode = mutableStateFlow.asLiveData(dispatcherProvider.Default)
+
+    val vaccinationCertificate = vaccinationRepository.vaccinationInfos
+        .map { persons ->
+            val findVaccinationDetails = findVaccinationDetails(persons)
+            generateQrCode(findVaccinationDetails.certificate)
+            findVaccinationDetails
+        }
+        .asLiveData(context = dispatcherProvider.Default)
 
     val errors = SingleLiveEvent<Throwable>()
     val events = SingleLiveEvent<VaccinationDetailsNavigation>()
 
-    fun deleteVaccination() = launch {
-        try {
-            Timber.d("deleteVaccination")
-            vaccinationRepository.deleteVaccinationCertificate(vaccinationCertificateId)
-            events.postValue(VaccinationDetailsNavigation.Back)
-        } catch (e: Exception) {
-            Timber.d(e, "deleteVaccinationCertificate failed")
-            errors.postValue(e)
-        }
-    }
+    fun onClose() = events.postValue(VaccinationDetailsNavigation.Back)
 
-    fun onClose() {
-        events.postValue(VaccinationDetailsNavigation.Back)
-    }
+    fun openFullScreen() = qrCodeText?.let { events.postValue(VaccinationDetailsNavigation.FullQrCode(it)) }
 
-    private fun findVaccinationDetails(vaccinatedPersons: Set<VaccinatedPerson>): VaccinationDetails {
-        val vaccinatedPerson = vaccinatedPersons.find { vaccinatedPerson ->
-            vaccinatedPerson
-                .vaccinationCertificates
-                .any { it.certificateId == vaccinationCertificateId }
+    private fun findVaccinationDetails(
+        vaccinatedPersons: Set<VaccinatedPerson>
+    ): VaccinationDetails {
+        val person = vaccinatedPersons.find { p ->
+            p.vaccinationCertificates.any { it.certificateId == vaccinationCertificateId }
         }
 
+        val certificate = person?.vaccinationCertificates?.find { it.certificateId == vaccinationCertificateId }
         return VaccinationDetails(
-            certificate = vaccinatedPerson?.vaccinationCertificates?.find {
-                it.certificateId == vaccinationCertificateId
-            },
-            isComplete = vaccinatedPerson?.getVaccinationStatus() == VaccinatedPerson.Status.COMPLETE
+            certificate = certificate,
+            isComplete = person?.getVaccinationStatus() == VaccinatedPerson.Status.COMPLETE,
```

Comment:
I think this needs to be changed to `person?.getVaccinationStatus() == VaccinatedPerson.Status.IMMUNITY` so that the background is only blue after 14 days of the final vaccination have passed.